### ISSUE TITLE
After recording, replay without saving fails (replays wrong macro)

### ISSUE THREAD

**Issue by teamdocs** [teamdocs/kantu2018#34]
- I record
- then do NOT save, 
 - then click "play",

=>  it replays something old, but not the macro that was just recorded

(Note: I am not sure if this is a regression issue or an old bug that I just found now... will check this later)

**Comment by teamdocs** [comment]
I upgraded this to blocker as the whole macro is lost. This is a regression issue, it works in earlier versions (e. g. V2.02).

At the very least we need to warn users if he plays a new macro before saving the old.

**Comment by kuznetsoffandrey** [comment]
Is this a bug you mean or not?
When I start recording (mouse mode), then I press Save and select "Dont save" - it continues recording
When I start recording (mouse mode), then press Stop - it drops me to the macros list

In what case it should work correct?

**Comment by teamdocs** [comment]
>When I start recording (mouse mode), then I press Save and select "Dont save" - it continues recording

Actually SAVE should not work during recording (minor)

>When I start recording (mouse mode), then press Stop - it drops me to the macros list

That is ok. My bug description was unclear:

-  I record
**- press STOP**
- then do NOT save the macro,
- then click "play",

**Comment by kuznetsoffandrey** [comment]
Let s check it once again. Maybe screencast will be more suitable
1. Record button pressed, I do some movements
2. Press stop
3. Macros list is opened (strange...) Seems it is more logical that the list of recorded operations is visible
4. Then I press save button
5. Select do not save
6. The saved script should be started (Now starts the one selected in the macros list)

**Comment by teamdocs** [comment]
Yes, a screencast makes good sense, please see: https://www.screencast.com/t/6qmpR0qGvGL (issue starts at 0:45s)

>Macros list is opened (strange...) Seems it is more logical that the list of recorded operations is visible

Agree!

>Then I press save button Select do not save
>The saved script should be started (Now starts the one selected in the macros list)

In this case, if someone on purpose cancels save, it is ok if the recording is lost. But in my case (see screencast) I press PLAY directly after recording (no save). Then I expect that the recorded macro is replayed.

**Comment by kuznetsoffandrey** [comment]
Seems now I got what the problem is. However it doesn't look like an obvious situation for user when the recorded command list disappears. But this can be a next feature

**Comment by teamdocs** [comment]
>However it doesn't look like an obvious situation for user when the recorded command list disappears. But this can be a next feature

Agree, this can be in the next milestone. For now, it will be enough if the macro is not lost if PLAY is used before SAVE. Maybe even a dialog box (Please save the macro before replaying it) will be ok.

**Comment by kuznetsoffandrey** [comment]
The bug is fixed. Now recorded commands are not lost after play button pushed
